api/controllers/v1/location/history.js

```diff
@@ -0,0 +1,30 @@
+const ControllerService = require('../../../services/ControllerService');
+const ErrorService = require('../../../services/ErrorService');
+
+module.exports = async (req, res) => {
+
+  try {
+    const locationsHPopulated = await HLocation.find({ id: req.params.id })
+    .populate('reviewer')
+    .populate('author')
+    .populate('entrance');
+
+    if (!locationsHPopulated) {
+      return res.notFound(`No history found for location ${req.params.id}`);
+    }
+
+    const params = {};
+    params.controllerMethod = 'GrottoController.findAll';
+    params.notFoundMessage = 'No location history found.';
```

Comment:
You can drop the line `params.notFoundMessage` --> we don't use it any more because the case where the resource is not found is handled just before. 

Instead, just do : 
```js
const params = {
   controllerMethod: 'GrottoController.findAll'
}
```